test/function/config/wel_supported.cpp

```diff
@@ -335,6 +335,47 @@ namespace {
     EXPECT_ERROR_ID_EQ( STUMPLESS_ARGUMENT_EMPTY );
   }
 
+  TEST_F( WelSupportedTest, InvalidUtf8InsertionString ) {
+    const struct stumpless_entry *entry_result;
+    LPCSTR invalid_utf8_string = "\xc3\x28 invalid";
+
+    entry_result = stumpless_set_wel_insertion_string( insertion_entry, 0, invalid_utf8_string );
+    EXPECT_NULL( entry_result );
```

Comment:
Add checks after calls with invalid UTF8 to make sure that the error id is set correctly: it should be `STUMPLESS_INVALID_ENCODING`. See some of the above tests for examples of how to do this using the `EXPECT_ERROR_ID_EQ` macro. Also note that this helper requires a `stumpless_error` struct named "error" to be declared in order to work.